src/app/FakeLib/UnitTest/NUnit/NUnit3.fs

```diff
@@ -0,0 +1,234 @@
+open System.IO
+open Fake
+
+type NUnit3ProcessModel = 
+    | DefaultProcessModel
+    | SingleProcessModel
+    | SeparateProcessModel
+    | MultipleProcessModel with 
+    member x.ParamString =
+        match x with
+        | DefaultProcessModel -> ""
+        | SingleProcessModel -> "Single"
+        | SeparateProcessModel -> "Separate" 
+        | MultipleProcessModel -> "Multiple"
+/// The /domain option controls of the creation of AppDomains for running tests. See [NUnit-Console Command Line Options](http://www.nunit.org/index.php?p=consoleCommandLine&r=2.6.4)
+type NUnit3DomainModel = 
+    /// The default is to use multiple domains if multiple assemblies are listed on the command line. Otherwise a single domain is used.
+    | DefaultDomainModel
+    /// No domain is created - the tests are run in the primary domain. This normally requires copying the NUnit assemblies into the same directory as your tests.
+    | NoDomainModel
+    /// A test domain is created - this is how NUnit worked prior to version 2.4
+    | SingleDomainModel
+    /// A separate test domain is created for each assembly
+    | MultipleDomainModel with
+    member x.ParamString =
+        match x with
+        | DefaultDomainModel -> ""
+        | NoDomainModel -> "None"
+        | SingleDomainModel -> "Single"
+        | MultipleDomainModel -> "Multiple"
+
+type NUnit3ErrorLevel = TestRunnerErrorLevel
+
+type NUnit3Params =
+    { /// The path to the NUnit3 console runner: `nunit3-console.exe`
+      ToolPath : string
+
+      /// NUnit console runner name. ( `nunit3-console.exe`)
+      ToolName : string
+
+      /// The name (or path) of a file containing a list of tests to run or explore, one per line.
+      Testlist : string
+
+      /// An expression indicating which tests to run. It may specify test names, classes, methods, 
+      /// catgories or properties comparing them to actual values with the operators ==, !=, =~ and !~. 
+      /// See [NUnit documentation](https://github.com/nunit/nunit/wiki/Test-Selection-Language) for a full description of the syntax.
+      Where : string
+
+      /// Name of a project configuration to load (e.g.: Debug)
+      Config : string
+
+      /// Controls how NUnit loads tests in processes. See [NUnit3ProcessModel](fake-testing-nunit3-nunit3processmodel.html)
+      ProcessModel : NUnit3ProcessModel
+
+      /// Number of agents that may be allowed to run simultaneously assuming you are not running inprocess.
+      /// If not specified, all agent processes run tests at the same time, whatever the number of assemblies.
+      /// This setting is used to control running your assemblies in parallel.
+      Agents : int option
+
+
+      /// Controls how NUnit loads tests in processes. See: [NUnit3ProcessModel](fake-testing-nunit3-nunit3domainmodel.html).
+      Domain : NUnit3DomainModel
+      
+      /// Allows you to specify the version of the runtime to be used in executing tests.
+      Framework : string
```

Comment:
I thought about it, but I am not sure about which values should be provided. NUnit documentation does not list these values as nicely as for NUnit3DomainModel or NUnit3ProcessModel. Instead, it accepts any value and checks at runtime that it is [available](https://github.com/nunit/nunit/blob/master/src/NUnitEngine/nunit.engine/RuntimeFramework.cs#L252). The best that can be done is listing some of them (e.g. mono, net-4.5, v4.0, 2.0, mono-4.0) and adding option for Other(string). Do you think this would be better?